frontend/src/types/cypress.d.ts

```diff
@@ -0,0 +1,15 @@
+// Fix these types which clashes with Cypress types. This is needed because we're loading Cypress types in some JS files.
```

Comment:
There's another attempt to achieve the same, but requires a lot of exclusion of files that load Cypress in our main `tsconfig.json` which looks even worse in my opinion. You can see that here. https://github.com/metabase/metabase/pull/28215/commits/ddff71cc8121d1086078d6b6df8012cbaa750065#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0

Reply:
Should we use built-in typescript types instead? https://docs.cypress.io/guides/tooling/typescript-support

Reply:
If you mean built-in Cypress types for TypeScript, then I'm already using that.

I use this in any `tsconfig.json` file inside the directories that contain only Cypress file.

```
"types": ["cypress", "@testing-library/cypress"],
```

As far as I understand, this [doc from Cypress](https://docs.cypress.io/guides/tooling/typescript-support), suggests that we create `tsconfig.json` **inside** the cypress folder (which is supposed to contain every cypress-related files, as opposed to us that put both Cypress and non-Cypress files in the same directory).

And this is the problem this file fixes. This file is supposed to be for the main app **not** the cypress file. Since types are already correct by those new `tsconfig.json` files. The problem this file fixes is that since we put both Cypress and non-Cypress file under the same directory e.g. `frontend/test`, by loading the main app files, we're also loading some Cypress files unintentionally, and that causes the Cypress types to get load into TypeScript causing the type conflicts in global names like `expect`, `describe` and whatnot.

Another solution is to exclude every Cypress file in our main app's tsconfig.json which  [was done here](https://github.com/metabase/metabase/commit/ddff71cc8121d1086078d6b6df8012cbaa750065#diff-b55cdbef4907b7045f32cc5360d48d262cca5f94062e353089f189f4460039e0), but IMO looks nastier and probably harder to maintain since the excluded files list might need to be updated.

So, I settled on the middle ground where I left the main app TypeScript also loads the Cypress file (so we will have the type conflicts between jest and Chai), but override Cypress type to point to jest (which we use in our unit test instead).

In an ideal world, I believed the better solution should be moving all cypress-related files under the same directory **and** not putting any non-Cypress-related files inside it e.g. putting unit tests elsewhere, or just creating a new directory called `cypress` and moving all cypress tests there. If we do that, we can remove this `cypress.d.ts` hack + reduce the tsconfig.json files to 2 (1 for the main app, another for Cypress, also another if you count the `tsconfig.base.json`). But that entails we update all the config files e.g. GitHub actions and potentially certain scripts that expect Cypress files to be where they are right now which I don't want to do at the moment. That could be the next milestone, but this improvement is already self-contained, and have an immediate effect on the DX overall. I've been using this myself for the past weeks by merging this PR into any PR I'm working on or reviewing, and it works pretty well for me.

Reply:
This isn't needed anymore because https://github.com/metabase/metabase/pull/28749 is merged.